core/capabilities/ccip/launcher/deployment.go

```diff
@@ -1,142 +1,45 @@
 package launcher
 
 import (
-	"fmt"
+	"golang.org/x/exp/maps"
 
 	cctypes "github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/types"
-
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
 	"go.uber.org/multierr"
-
-	ccipreaderpkg "github.com/smartcontractkit/chainlink-ccip/pkg/reader"
 )
 
-// activeCandidateDeployment represents a active-candidate deployment of OCR instances.
-type activeCandidateDeployment struct {
-	// active is the active OCR instance.
-	// active must always be present.
-	active cctypes.CCIPOracle
-
-	// candidate is the candidate OCR instance.
-	// candidate may or may not be present.
-	// candidate must never be present if active is not present.
-	candidate cctypes.CCIPOracle
-}
-
-// ccipDeployment represents active-candidate deployments of both commit and exec
-// OCR instances.
-type ccipDeployment struct {
-	commit activeCandidateDeployment
-	exec   activeCandidateDeployment
-}
-
-// Close shuts down all OCR instances in the deployment.
-func (c *ccipDeployment) Close() error {
-	// we potentially run into this situation when
-	// trying to close an active instance that doesn't exist
-	// this check protects us from nil pointer exception
+type ccipPlugins map[ocrtypes.ConfigDigest]*cctypes.CCIPOracle
 
+// Close is used to shut down an entire don immediately
+func (c *ccipPlugins) Close() error {
```

Comment:
[Maps are already reference types in Go](https://go.dev/blog/maps#:~:text=Map%20types%20are%20reference%20types%2C%20like%20pointers%20or%20slices%2C%20and%20so%20the%20value%20of%20m%20above%20is%20nil%3B%20it%20doesn%E2%80%99t%20point%20to%20an%20initialized%20map), I think you can avoid the pointer receiver here which causes awkward dereferencing in the code.

```suggestion
func (c ccipPlugins) Close() error
```